extension/modules/personas.js

```diff
@@ -0,0 +1,1480 @@
+		}
+		else{
+			var toolbarbutton=PersonaController.createToolbarButton(document);
+			(document.getElementById("navigator-toolbox") || document.getElementById("mail-toolbox")).palette.appendChild(toolbarbutton);
+			//PersonaController.placeToolbarButton(window,toolbarbutton);
+			if(PersonaController.appInfo.ID!="{8de7fcbb-c55c-4fbe-bfc5-fc555c87dbc4}") PersonaController.placeToolbarButton(window,toolbarbutton);
+			else{
+				window.setTimeout(function(){
+					PersonaController.placeToolbarButton(window,toolbarbutton);
+				},1000)					
+			}				
+		}
+		window.addEventListener("aftercustomization", PersonaController.afterCustomization, false);				
+	},
+	removeToolbarButton:function(window){
+		if (window.CustomizableUI) {
+			window.CustomizableUI.destroyWidget("personas-toolbar-button");
+		}
+		else{
+			var document = window.document;
+			var buttonId="personas-toolbar-button";
+			var button=document.getElementById(buttonId);
+			button.parentNode.removeChild(button);
+		}
+		window.removeEventListener("aftercustomization", PersonaController.afterCustomization, false);
+		PersonaController.widgetCreated=false;
+	},
+	placeToolbarButton:function(window,toolbarbutton){
+		if(!PersonaController._prefs.get("tbinsert")) return;
+		var document=window.document;
+		if (window.CustomizableUI) {
+			window.CustomizableUI.addWidgetToArea("personas-toolbar-button", window.CustomizableUI.AREA_NAVBAR,5);
+			PersonaController._prefs.set("tbplaced",true);
+		}
+		else{
+			var toolbar = document.querySelector("[currentset^='" + toolbarbutton.id + ",'],[currentset*='," + toolbarbutton.id +
+			",'],[currentset$='," + toolbarbutton.id + "']");
+			if(toolbar){
+				var currentset=toolbar.getAttribute("currentset").split(",");
+				var i=currentset.indexOf(toolbarbutton.id) + 1;
+				var before=null;
+				while (i < currentset.length && !(before=document.getElementById(currentset[i]))) i++;
+				toolbar.insertItem(toolbarbutton.id, before, null, false);	
+			}
+			else{		
+				var navbar = (document.getElementById("nav-bar") || document.getElementById("mail-bar3") || document.getElementById("msgToolbar"));
+				navbar.insertItem(toolbarbutton.id, null, null, false);			
+			}
+			if(!document.getElementById(toolbarbutton.id)) (toolbar || navbar).insertItem(toolbarbutton.id, null, null, false);				
+		}
+	},
+	afterCustomization:function(event){
+		var window=event.currentTarget;
+		var document=window.document;
+		var ythdprefsinstance = Components.classes["@mozilla.org/preferences-service;1"].getService(Components.interfaces.nsIPrefBranch);
+		if(!document.getElementById("personas-toolbar-button") || document.getElementById("personas-toolbar-button").parentNode.id=="BrowserToolbarPalette") ythdprefsinstance.setBoolPref("extensions.personas.tbinsert",false);
+		else ythdprefsinstance.setBoolPref("extensions.personas.tbinsert",true);
+	},
+	addToolsMenu:function(window){
+		var document=window.document;
```

Comment:
OK, but some users remove Toolbar button to preserve space and only use Tools menu. As of now, current version of Personas Plus on AMO allows that. When Tools menu removed, they may not be able to use the add-on if the toolbar button is removed by user.

What do you think?